packages/terra-list/src/ListItem.jsx

```diff
@@ -22,57 +28,73 @@ const propTypes = {
    */
   isSelectable: PropTypes.bool,
   /**
-   * Whether or not the list item has a disclosure indicator presented.
+   * The associated metaData to be provided in the onSelect callback.
    */
-  hasChevron: PropTypes.bool,
+  metaData: PropTypes.object,
+  /**
+   * Function callback for when the appropriate click or key action is performed.
+   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
+   */
+  onSelect: PropTypes.func,
   /**
    * Function callback for the ref of the li.
    */
   refCallback: PropTypes.func,
 };
+/* eslint-enable react/forbid-prop-types */
 
 const defaultProps = {
-  content: undefined,
+  children: [],
+  hasChevron: undefined,
   isSelected: false,
   isSelectable: undefined,
-  hasChevron: undefined,
 };
 
 const ListItem = ({
-  content,
+  children,
+  hasChevron,
   isSelected,
   isSelectable,
-  hasChevron,
+  metaData,
+  onBlur,
+  onClick,
+  onKeyDown,
+  onMouseDown,
+  onSelect,
   refCallback,
   ...customProps
 }) => {
   const listItemClassNames = cx([
     'list-item',
     { selected: isSelected },
     { 'is-selectable': isSelectable },
+    { 'item-has-chevron': hasChevron },
     customProps.className,
   ]);
 
-  const ariaSpread = isSelectable ? { role: 'option', 'aria-selected': isSelected } : {};
+  const ariaSpread = {};
+  if (isSelectable) {
+    ariaSpread.onClick = ListUtils.wrappedOnClickForItem(onClick, onSelect, metaData);
+    ariaSpread.onKeyDown = ListUtils.wrappedOnKeyDownForItem(onKeyDown, onSelect, metaData);
+    ariaSpread.tabIndex = '0';
+    ariaSpread.role = 'option';
+    ariaSpread['aria-selected'] = isSelected;
+    ariaSpread['data-item-show-focus'] = 'true';
+    ariaSpread.onBlur = ListUtils.wrappedOnBlur(onBlur, event => event.currentTarget.setAttribute('data-item-show-focus', 'true'));
+    ariaSpread.onMouseDown = ListUtils.wrappedOnMouseDown(onMouseDown, event => event.currentTarget.setAttribute('data-item-show-focus', 'false'));
+  }
 
+  let childContent = children;
   if (hasChevron) {
-    const chevron = <span className={cx('chevron')}><ChevronRight height="0.8em" width="0.8em" /></span>;
-
-    return (
-      <li {...customProps} {...ariaSpread} className={listItemClassNames} ref={refCallback}>
-        <Arrange
-          fill={content}
-          fitEnd={chevron}
-          align="center"
-          fillAttributes={{ className: cx('arrange-content') }}
-        />
-      </li>
-    );
+    childContent = [
```

Comment:
Should we check if children is empty? Or is the desired behavior to still render a list item with a chevron if no children were passed?

Reply:
I don't know that we need to care if they pass empty content. I don't think it'll hurt anything.